### DISCUSSION TITLE
Suggestions for the User Token Guide

### DISCUSSION THREAD

**Discussion by lmc395** [Tyrrrz/DiscordChatExporter#984] [General]
First off: sorry if the wiki's not open for feedback. If it is, though, I've got some suggestions for the documentation on finding user tokens:

 * "Through your web browser"
   * Add directions for finding your token using the network monitor on Firefox.
   * Incorporate the directions under "I can't find/copy my token!" in `Troubleshooting.md`.
   * Reorganize the information into something like the following:
     * In Chrome
       * Using the network Monitor
       * Using the storage inspector
     * In Firefox
       * Using the network monitor
       * Using the storage inspector
 * "Through the desktop app"
   * Add a table to help users find their settings.json file.
   * Add another table to help BetterDiscord users find their settings.json.
   * Edit the the hyperlink in step five to point to the "In Chrome" section.

Putting all that into action, `Obtaining-Token-and-Channel-IDs.md`'d look something like this:

>  . . . 
> ### How to get a User Token
> 
> > Automating user accounts is technically against TOS, use at your own risk.
> 
> #### Through your web browser
> 
> ##### In Chrome
> 
> ###### Using the network monitor
> 
> 1. Open the developer tools by pressing **Ctrl+Shift+I** (⌘⌥I on a Mac) on Discord  
> 2. Go to the **Network** tab  
> 3. Press **Ctrl+R** (⌘R) to reload  
> 4. Search for `messages?limit=50` and click it  
> 5. On the right, make sure the Headers tab is selected  
> 6. Scroll down to the Request Headers section and look for the `authorization` code
> 
> Too complicated? [This image might help you](https://user-images.githubusercontent.com/21324788/44003408-13eb2088-9e07-11e8-9c3c-016d9876bbea.png).    
> Special thanks to [@llamasking](https://github.com/Tyrrrz/DiscordChatExporter/issues/76#issuecomment-412349261) for providing the screenshot.
> 
> ###### Using the storage inspector
> 
> 1. Open Discord in your **web browser** and login  
> 2. Press **Ctrl+Shift+I** (⌥⌘I on macOS) to open **developer tools**  
> 3. Press **Ctrl+Shift+M** (⇧⌘M) to toggle **device toolbar**  
> 4. Navigate to the **Application** tab  
> 5. On the left, expand **Local Storage** and select **https꞉//discord.com** 
> 6. Type `token` into the **Filter** box  
> 7. If the **token** key does not appear, press **Ctrl+R** (⌘R) to reload  
> 8. Copy the **value** of the token key  
> 
> ##### In Firefox
> 
> ###### Using the network monitor
> 
> 1. Using Firefox, navigate to discord.com and login.
> 2. Press <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>E</kbd> (or <kbd>Cmd</kbd>+<kbd>Opt</kbd>+<kbd>E</kbd> on macOS). Firefox’s [web developer tools](https://firefox-source-docs.mozilla.org/devtools-user/) will display at the bottom of the window, and the [network monitor](https://firefox-source-docs.mozilla.org/devtools-user/network_monitor/) will display.
> 3. Press <kbd>F5</kbd>. The page will reload, and the [network request list](https://firefox-source-docs.mozilla.org/devtools-user/network_monitor/request_list/index.html) will populate with entries.
> 4. Type `library` into the filter. The network request list will filter out any entries not containing the string `library`.
> 5. Click `library`. The [network request details pane](https://firefox-source-docs.mozilla.org/devtools-user/network_monitor/request_details/index.html) will display. The [headers tab](https://firefox-source-docs.mozilla.org/devtools-user/network_monitor/request_details/index.html#network-monitor-request-details-headers-tab) should be active by default. If it isn’t, click it.
> 6. Type `authorization` into the text box labelled `Filter Headers`.
> 7. Scroll down until you see an entry labeled [authorization](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization) under `Request Headers`.
> 8. Right-click the entry labeled `authorization` and select `copy value`.
> 
> ###### Using the storage inspector
> 
> 1. Visit https://discord.com/login and login  
> 2. Press **F12**(or Ctrl+Shift+I) to open **Web Developer Tools**, which is in **Application Menu\More Tools** with the mouse
> 3. Press **Ctrl+Shift+M** to toggle **Responsive Design Mode**  , which is a button at upper right of **Web Developer Tools** with the mouse  
> (3.1. It is like on mobile. It may be useful to change the selection of **Device Settings** at left of top bar such as iPhone)  
> 4. Navigate to the **Storage** tab  
> 5. On the left, expand **Local Storage** and select **https꞉//discord.com**  
> 6. Type `token` into the **Filter** box  
> 7. If the **token** key does not appear, press **Ctrl+R** to reload, or check step 3/3.1  
> 8. Copy the **value** of the token key, looked like "xODk3NTk4NpeB7jMwMTYzMTU.43QYjx.4k4h_H-i8feb_rY5OTk07bYv9xI" while the "" does not matter
>  . . . 
> #### Through the desktop app
> 1. Find and open `settings.json` in your preferred text editor.
> 
>     | OS      | Stable                                                | Canary                                                      | Public Test Beta                                         |
>     |---------|-------------------------------------------------------|-------------------------------------------------------------|----------------------------------------------------------|
>     | Windows | `%APPDATA%\discord\settings.json`                     | `%APPDATA%\discordcanary\settings.json`                     | `%APPDATA%\discordptb\settings.json`                     |
>     | macOS   | `~/Library/Application Support/discord/settings.json` | `~/Library/Application Support/discordcanary/settings.json` | `~/Library/Application Support/discordptb/settings.json` |
>     | Linux   | `~/.config/discord/settings.json`                     | `~/.config/discordcanary/settings.json`                     | `~/.config/discordptb/settings.json`                     |
> 
>     Note: If you use BetterDiscord, use the following table instead.
> 
>     | OS      | Stable                                                                  | Canary                                                                  | Public Test Beta                                                     |
>     |---------|-------------------------------------------------------------------------|-------------------------------------------------------------------------|----------------------------------------------------------------------|
>     | Windows | `%APPDATA%\BetterDiscord\data\stable\settings.json`                     | `%APPDATA%\BetterDiscord\data\canary\settings.json`                     | `%APPDATA%\BetterDiscord\data\ptb\settings.json`                     |
>     | macOS   | `~/Library/Application Support/BetterDiscord/data/stable/settings.json` | `~/Library/Application Support/BetterDiscord/data/canary/settings.json` | `~/Library/Application Support/BetterDiscord/data/ptb/settings.json` |
>     | Linux   | `~/.config/BetterDiscord/data/stable/settings.json`                     | `~/.config/BetterDiscord/data/canary/settings.json`                     | `~/.config/BetterDiscord/data/ptb/settings.json`                     |
>  . . . 
> 5. [Follow either instructions for finding your token on Chrome](https://github.com/Tyrrrz/DiscordChatExporter/wiki/Obtaining-Token-and-Channel-IDs#in-chrome).
>  . . . 

Attached is an edit of `Obtaining-Token-and-Channel-IDs.md` with the above changes: [Obtaining-Token-and-Channel-IDs-a.md](https://github.com/Tyrrrz/DiscordChatExporter/files/10368006/Obtaining-Token-and-Channel-IDs-a.md)

That all said, hopefully this isn't overbearing, but I'd like to pitch a rewrite for the above:

> 
> ### How to get a User Token
> 
> **Caution**: [Automating user accounts violates Discord's terms of service](https://support.discord.com/hc/en-us/articles/115002192352-Automated-user-accounts-self-bots-) and may result in account termination. Use at your own risk.
> 
> #### Through your web browser
> 
> Prerequisite step: Navigate to discord.com and login.
> 
> ##### In Chrome
> 
> ###### Using the network monitor
> 
> 1. <img width="500" align="right" src="https://i.imgur.com/zdDwIT5.jpg" />Press <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>I</kbd> (or <kbd>⌘</kbd>+<kbd>⌥</kbd>+<kbd>I</kbd> on macOS). Chrome's [DevTools](https://developer.chrome.com/docs/devtools/overview) tools will display.
> 
> <br clear="right" />
> <br />
> 
> 2. <img width="500" align="right" src="https://i.imgur.com/BDeG0zg.png" />Click the `Network` tab. The [network panel](https://developer.chrome.com/docs/devtools/overview/#network) will open
> 
> <br clear="right" />
> <br />
> 
> 3. <img width="500" align="right" src="https://i.imgur.com/0Lgj0vk.png" />Press <kbd>F5</kbd>. The page will reload, and the network log (the lower half of the network panel) will display several entries.
> 
> <br clear="right" />
> <br />
> 
> 4. <img width="500" align="right" src="https://i.imgur.com/rnZG8Id.png" />Click the text box labelled `Filter` and type `library`. The entries will filter down to a single entry named `library`.
> 
> <br clear="right" />
> <br />
> 
> 5. <img width="500" align="right" src="https://i.imgur.com/29dE3fR.png" />Click the entry named `library`. A panel will open to the right and display details about the entry. Click the `Headers` tab if it isn't already active.
> 
> <br clear="right" />
> <br />
> 
> 6. <img width="500" align="right" src="https://i.imgur.com/u7CxXAt.png" />Scroll through the contents of the `Headers` tab until you find an entry beginning with `authorization:`.
> 
> <br clear="right" />
> <br />
> 
> 7. <img width="500" align="right" src="https://i.imgur.com/dXcXzma.png" />Right-click the entry and click `copy value`.
> 
> <br clear="right" />
> <br />
> 
> ###### Using the storage inspector
> 
> 1. <img width="500" align="right" src="https://i.imgur.com/zdDwIT5.jpg" />Press <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>I</kbd> (or <kbd>⌘</kbd>+<kbd>⌥</kbd>+<kbd>I</kbd> on macOS). Chrome's [DevTools](https://developer.chrome.com/docs/devtools/overview/) will display.
> 
> <br clear="right" />
> <br />
> 
> 2. <img width="500" align="right" src="https://i.imgur.com/biAUIop.png" />Press <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>M</kbd> (or <kbd>⌘</kbd>+<kbd>Shift</kbd>+<kbd>M</kbd>). Chrome will enter [Device Mode](https://developer.chrome.com/docs/devtools/device-mode/), and the webpage will display as if on a mobile device.
> 
> <br clear="right" />
> <br />
> 
> 3. <img width="500" align="right" src="https://i.imgur.com/oUDRZoy.png" />If necessary, click the `>>` at the right end of the tab bar, and click `Application`. The [application panel](https://developer.chrome.com/docs/devtools/overview/#application) will display.
> 
> <br clear="right" />
> <br />
> 
> 4. <img width="500" align="right" src="https://i.imgur.com/sydNPia.png" />In the menu to the right, under `Storage`, expand `Local Storage` if necessary, then click `https://discord.com`. The pane to the right will display a list of key-value pairs.
> 
> <br clear="right" />
> <br />
> 
> 5. <img width="500" align="right" src="https://i.imgur.com/qKo0ny9.png" />In the text box marked `Filter`, type `token`. The entries will filter down to those containing the string `token`.
> 
> <br clear="right" />
> <br />
> 
> 6. <img width="500" align="right" src="https://i.imgur.com/caj3lQq.png" />Click the entry `token`. (Note: if the token doesn't display, try refreshing by pressing <kbd>F5</kbd>.)
> 
> <br clear="right" />
> <br />
> 
> 7. <img width="500" align="right" src="https://i.imgur.com/SwWFIH4.png" />Click the text box at the bottom, press <kbd>Ctrl</kbd>+<kbd>A</kbd> (or <kbd>⌘</kbd>+<kbd>A</kbd> on macOS) then <kbd>Ctrl</kbd>+<kbd>C</kbd> (or <kbd>⌘</kbd>+<kbd>C</kbd> on macOS) to copy the value to your clipboard.
> 
> <br clear="right" />
> <br />
> 
> ##### In Firefox
> 
> ###### Using the network monitor
> 
> 1. <img width="500" align="right" src="https://i.imgur.com/O34nwdG.png" />Press <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>E</kbd> (or <kbd>⌘</kbd>+<kbd>⌥</kbd>+<kbd>E</kbd> on macOS). Firefox’s [web developer tools](https://firefox-source-docs.mozilla.org/devtools-user/) will display at the bottom of the window, and the [network monitor](https://firefox-source-docs.mozilla.org/devtools-user/network_monitor/) will display.
> 
> <br clear="right" />
> <br />
> 
> 2. <img width="500" align="right" src="https://i.imgur.com/j00QzhU.png" />Press <kbd>F5</kbd>. The page will reload, and the [network request list](https://firefox-source-docs.mozilla.org/devtools-user/network_monitor/request_list/index.html) will populate with entries.
> 
> <br clear="right" />
> <br />
> 
> 3. <img width="500" align="right" src="https://i.imgur.com/efUCfBO.png" />Type `library` into the filter. The network request list will filter out any entries not containing the string `library`.
> 
> <br clear="right" />
> <br />
> 
> 4. <img width="500" align="right" src="https://i.imgur.com/cdJZ7Q1.png" />Click `library`. The [network request details pane](https://firefox-source-docs.mozilla.org/devtools-user/network_monitor/request_details/index.html) will display. The [headers tab](https://firefox-source-docs.mozilla.org/devtools-user/network_monitor/request_details/index.html#network-monitor-request-details-headers-tab) should be active by default. If it isn’t, click it.
> 
> <br clear="right" />
> <br />
> 
> 5. <img width="500" align="right" src="https://i.imgur.com/zBmq1JW.png" />Type `authorization` into the text box labelled `Filter Headers`.
> 
> <br clear="right" />
> <br />
> 
> 6. <img width="500" align="right" src="https://i.imgur.com/O3blcIS.png" />Scroll down until you see an entry labeled [authorization](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization) under `Request Headers`.
> 
> <br clear="right" />
> <br />
> 
> 7. <img width="500" align="right" src="https://i.imgur.com/zHYEYoZ.png" />Right-click the entry labeled `authorization` and select `copy value`.
> 
> <br clear="right" />
> <br />
> 
> ###### Using the storage inspector
> 
> 1. <img width="500" align="right" src="https://i.imgur.com/A8jqpkm.png" />Press <kbd>Shift</kbd>+<kbd>F9</kbd>. Firefox’s [web developer tools](https://firefox-source-docs.mozilla.org/devtools-user/) will display at the bottom of the window, and the [storage](https://firefox-source-docs.mozilla.org/devtools-user/storage_inspector/index.html) panel will be selected.
> 
> <br clear="right" />
> <br />
> 
> 2. <img width="500" align="right" src="https://i.imgur.com/TGcbB7f.png" />Press <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>M</kbd> (or <kbd>⌘</kbd>+<kbd>⌥</kbd>+<kbd>M</kbd> on macOS). Firefox will toggle [responsive design mode](https://firefox-source-docs.mozilla.org/devtools-user/responsive_design_mode/), and the web page will display as if on a mobile device. (Note: Discord may steal focus and respond to the command by toggling mute. If this happens, return focus to Firefox’s web developer tools by clicking somewhere in it, then try the command again.)
> 
> <br clear="right" />
> <br />
> 
> 3. <img width="500" align="right" src="https://i.imgur.com/2xWkep9.png" />In the [storage tree](https://firefox-source-docs.mozilla.org/devtools-user/storage_inspector/index.html#storage-inspector-storage-tree) (the list on the left side of the web developer tools panel), click [Local Storage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage). The entry will expand, and the entry `https://discord.com` will display beneath it.
> 
> <br clear="right" />
> <br />
> 
> 4. <img width="500" align="right" src="https://i.imgur.com/tGlGuOL.png" />In the storage tree, click `https://discord.com`. The [table widget](https://firefox-source-docs.mozilla.org/devtools-user/storage_inspector/index.html#storage-inspector-table-widget) to the right of the storage tree will display several key-value pairs.
> 
> <br clear="right" />
> <br />
> 
> 5. <img width="500" align="right" src="https://i.imgur.com/hDNsnZ5.png" />In the text box labelled `Filter items` at the top of the table widget, enter `token`. The table will now only display entries containing the string `token`.
> 
> <br clear="right" />
> <br />
> 
> 6. <img width="500" align="right" src="https://i.imgur.com/8fKId1W.png" />Click the entry `token`. The [sidebar](https://firefox-source-docs.mozilla.org/devtools-user/storage_inspector/index.html#storage-inspector-sidebar) will display. (Note: If the token doesn’t display, try refreshing by pressing <kbd>F5</kbd>.)
> 
> <br clear="right" />
> <br />
> 
> 7. <img width="500" align="right" src="https://i.imgur.com/yD1ZuR9.png" />Right-click the single entry in the sidebar and select `copy`.
> 
> <br clear="right" />
> <br />
> 
> #### Through the desktop app / Enabling web developer tools
> 
> ##### By editing the settings file
> 
> 1. If Discord is running, exit the application by right-clicking the icon and clicking `Quit Discord`.
> 
> 2. Open Discord's settings file in your preferred text editor. See the following table for help finding it:
> 
>     | OS      | Stable                                                | Canary                                                      | Public Test Beta                                         |
>     |---------|-------------------------------------------------------|-------------------------------------------------------------|----------------------------------------------------------|
>     | Windows | `%APPDATA%\discord\settings.json`                     | `%APPDATA%\discordcanary\settings.json`                     | `%APPDATA%\discordptb\settings.json`                     |
>     | macOS   | `~/Library/Application Support/discord/settings.json` | `~/Library/Application Support/discordcanary/settings.json` | `~/Library/Application Support/discordptb/settings.json` |
>     | Linux   | `~/.config/discord/settings.json`                     | `~/.config/discordcanary/settings.json`                     | `~/.config/discordptb/settings.json`                     |
> 
>     If you use BetterDiscord, use the following table instead:
> 
>     | OS      | Stable                                                                  | Canary                                                                  | Public Test Beta                                                     |
>     |---------|-------------------------------------------------------------------------|-------------------------------------------------------------------------|----------------------------------------------------------------------|
>     | Windows | `%APPDATA%\BetterDiscord\data\stable\settings.json`                     | `%APPDATA%\BetterDiscord\data\canary\settings.json`                     | `%APPDATA%\BetterDiscord\data\ptb\settings.json`                     |
>     | macOS   | `~/Library/Application Support/BetterDiscord/data/stable/settings.json` | `~/Library/Application Support/BetterDiscord/data/canary/settings.json` | `~/Library/Application Support/BetterDiscord/data/ptb/settings.json` |
>     | Linux   | `~/.config/BetterDiscord/data/stable/settings.json`                     | `~/.config/BetterDiscord/data/canary/settings.json`                     | `~/.config/BetterDiscord/data/ptb/settings.json`                     |
> 
> 3. Insert a blank line after the first curly bracket (`{`), add the text `"DANGEROUS_ENABLE_DEVTOOLS_ONLY_ENABLE_IF_YOU_KNOW_WHAT_YOURE_DOING": true,` to it, and save the file. Your file should resemble the following:
> 
> ```json
> {
>   "DANGEROUS_ENABLE_DEVTOOLS_ONLY_ENABLE_IF_YOU_KNOW_WHAT_YOURE_DOING": true,
>   "BACKGROUND_COLOR": "#202225",
>   "IS_MAXIMIZED": true
> }
> ```
> 
> 4. Launch Discord.
> 
> 5. To find your user token, continue [here](https://github.com/Tyrrrz/DiscordChatExporter/wiki/Obtaining-Token-and-Channel-IDs#in-chrome).
> 
> ##### Via settings menu (BetterDiscord only)
> 
> 1. <img width="500" align="right" src="https://i.imgur.com/mu1g4OF.png" />Click the User Settings button (the gear icon to the right of your username). Discord’s settings page will open.
> 
> <br clear="right" />
> <br />
> 
> 2. <img width="500" align="right" src="https://i.imgur.com/qFrIKON.png" />In the sidebar to the left, click `Settings` under the `BetterDiscord` group. BetterDiscord’s settings page will display.
> 
> <br clear="right" />
> <br />
> 
> 3. <img width="500" align="right" src="https://i.imgur.com/48Re5kj.png" />In the main panel to the right, expand the `Developer Settings` group if necessary, and toggle `DevTools` to enabled.
> 
> <br clear="right" />
> <br />
> 
> 4. Press <kbd>Esc</kbd>. The settings page will close.
> 
> 5. To find your user token, continue [here](https://github.com/Tyrrrz/DiscordChatExporter/wiki/Obtaining-Token-and-Channel-IDs#in-chrome).

Attached is an edit of `Obtaining-Token-and-Channel-IDs.md` with the above changes: [Obtaining-Token-and-Channel-IDs-b.md](https://github.com/Tyrrrz/DiscordChatExporter/files/10368007/Obtaining-Token-and-Channel-IDs-b.md) (Heads up, by the way: the images in this one use alignment tags that don't work on mobile browsers. Rather than floating to right, the images'll display inline before the text on mobile.)

This all might seem a bit excessive (especially considering what's already in the wiki does a better job than most dedicated guides), but I think these changes are worth while. More detailed, clear, well-structured instructions with images'll be more intuitive to users-- and it's worth considering Discord isn't exclusively used by the kinds of people who'll feel comfortable using "web developer tools," and wanting your chat logs exported isn't a niche ask. More complete instructions'll cover more users and give them more alternatives if they run into issues. Polishing things up (formatting, structure, etc) and linking to authoritative sources'll help earn the user's trust (which I'm guessing is pretty important given user tokens're involved).

**Comment by Tyrrrz** [comment]
Thanks for the suggestions, @lmc395. I temporarily unlocked wiki for editing so that you can edit the pages yourself. It's usually restricted to prevent harmful edits.

**Reply by lmc395** [reply]
Thanks! I've updated the page on Obtaining Token and Channel IDs.

**Reply by Tyrrrz** [reply]
Thank you!
